submissions/jeff-jeremy/scripts/article.js

```diff
@@ -0,0 +1,191 @@
+        if (callback) callback();
+      }
+    );
+  };
+
+  // DONE: Refactor to expect the data from the database, rather than localStorage.
+  Article.loadAll = function(rows) {
+    Article.all = rows.map(function(ele) {
+      return new Article(ele);
+    });
+  };
+
+  Article.fetchAll = function(next) {
+    /* TODO: Refactor the .fetchAll() method to check
+        if the database holds any records or not.
+
+      If the DB has data already, we'll load up the data
+        (most recent article first!), and then hand off control to the View.
+
+      If the DB is empty, we need to retrieve the JSON and process it. */
+    webDB.execute('SELECT * FROM articles;', function(rows) { // TODO: fill these quotes to 'select' our table.
+      if (rows.length) {
+        /* TODO:
+           1 - Use Article.loadAll to instanitate these rows,
+           2 - Pass control to the view by calling the next function that
+                was passed in to Article.fetchAll */
+        Article.loadAll(rows);
+        next();
+
+      } else {
+        $.getJSON('/data/hackerIpsum.json', function(data) {
+          // Save each article from this JSON file, so we don't need to request it next time:
+          data.forEach(function(obj) {
+            var article = new Article(obj); // This will instantiate an article instance based on each article object from our JSON.
+            /* TODO:
+               1 - 'insert' the newly-instantiated article in the DB:
+                (hint: what can we call on each 'article' instance?). */
+            article.insertRecord();
+            // webDB.execute(
+            //   [
+            //     {
+            //       'sql': 'INSERT INTO articles (title, category, author, authorUrl, publishedOn, body) VALUES (?, ?, ?, ?, ?, ?);',
+            //       'data': [this.title, this.category, this.author, this.authorUrl, this.publishedOn, this.body]
+            //     }
+            //   ]
+            // );
+
+          });
+          // Now get ALL the records out the DB, with their database IDs:
+          webDB.execute('SELECT * FROM articles;', function(rows) { // TODO: select our now full table
+            // TODO:
+            // 1 - Use Article.loadAll to instanitate these rows,
+            // 2 - Pass control to the view by calling the next function that was passed in to Article.fetchAll
+            Article.loadAll(rows);
+            next();
+
+          });
+        });
+      }
+    });
+  };
+
+  Article.prototype.insertRecord = function(callback) {
+    webDB.execute(
+      [
+        {
+          // TODO: Insert an article instance into the database:
+          // Note: this method will be called elsewhere after we retrieve our JSON
+          'sql': 'INSERT INTO articles (title, category, author, authorUrl, publishedOn, body) VALUES (?,?,?,?,?,?)',
+          'data': [this.title, this.author, this.authorUrl, this.category, this.publishedOn, this.body]
```

Comment:
check the order you're passing in these properties in the data array,
they need to match the order of the fields being passed in your sql query